acceptance/tests/agent/agent_can_reconnect_after_master_failure.rb

```diff
@@ -0,0 +1,50 @@
+test_name 'C97323: agents can reconnect after master crash/restart' do
+
+  logfiles = {}
+  agents.each do |agent|
+    logfiles[agent.hostname] = agent.tmpfile(File.basename(__FILE__, '.*'))
+    on(agent, puppet_resource("user puppet ensure=present"))
```

Comment:
If you specify `--user agent['user']` then I think (but am not 100% sure), you won't have to create/remove the puppet user. The agent will chown the log file that the agent is running as.